### DISCUSSION TITLE
Allow for quick filter reset

### DISCUSSION THREAD

**Discussion by alexzeitgeist** [astubenbord/paperless-mobile#212] [Feature Requests]
Hi,

It's incredibly easy to filter content - for example by correspondent - with a single touch. 👍 It is, however, not as easy to remove the filter again. I would first instinctively swipe the screen back (swipe to the right), but that would only exit the application. It seems there is no context of "before" when the filter was applied. As an alternative, a single reset button would be helpful too.

As it is right now, I need to click on the filter button, and hit "Reset" to get back to the unfiltered list.

Cheers,
Alex

**Comment by astubenbord** [comment]
Hi, would it be sufficient to just have a small button above the filter button to reset it? I've had the same thought a while back, but I didn't think it was necessary.

**Reply by JigSawFr** [reply]
Can be fine at least to try with a button :)

**Comment by lagset** [comment]
Maybe an alternative to consider:

When activating a filter a temporary snack bar appears at the bottom, with content "Added filter <filtername> to active filters (<active filters count>)" or something like that. 

In the snack bar, there is a undo Button to remove the newly added filter.

This would improve transparency of the system (what has been set?) and the user would be able to quickly unset the last set filter.

It does, of course not solve the issue to quickly reset *all* active filters, but imho I don't think this needs to be a "one-tap" always accessible action.

**Reply by astubenbord** [reply]
That's also a good idea, especially when accidentally clicking on the correspondent in the list.

**Comment by astubenbord** [comment]
@alexzeitgeist the button is implemented in the newest beta, maybe give it a try and see if it works for you. If it's fine I'll release it for everyone.

**Reply by alexzeitgeist** [reply]
Hello, thanks for the quick implementation. It works fine. Not sure so much about the UX design aspect of it.

But I must apologize. Only after posting this discussion thread did I find out that you *can* quickly remove a filter. When you are in the filtered list, click on one of the filtered items, a tag or a correspondent for example, and the filter will disappear while leaving the other filters in place. Excellent 👌

**Reply by astubenbord** [reply]
Yeah, I'm also not the biggest fan of it. The official slack app has a similar layout for the buttons, but two FABs are not that common I suppose.

I'll think of a better way, maybe the quick toggles are sufficient.

**Comment by marvinwankersteen** [comment]
I have also missed the reset of the filters in the past and had the idea that you can reset the filter with "Press and hold" on the filter button.
The way it is currently, with an extra button, it is definitely better, but somehow it looks a bit out of place.
The function of the "press and hold" should then also be apparent, I think. So without optical change, the function is then probably lost.
